### PR TITLE
Update to channel list feature

### PR DESCRIPTION
**Pull request by dkasipovic**
So, as we discussed before, here are the changes I am proposing (feel free to reject, just showing an example).

Two changes here:

1. Instead of using `[1, 2, 3]` for channelButtons, you can now use `[ { number: 1, title: "Sometitle"}, ... ]` and buttons will have `title` as title instead of `Channel: 1`, `Channel: 2`, etc.

2. On connect to tv, get channel list and store it in `this.channelMap` as `number=>id` map. Later, when switching channel, if `this.channelMap[channelNumber]` exists, try to change channel by id instead of by number (fixed the channel switches not working for me).

Let me know what you think

### CODE REVIEW

**Comment by merdok** [comment]
Thanks for the pull request. I am still not convinced for this solution but i will have a look at your changes.

File: index.js [hunk 1]
```diff
@@ -66,6 +66,10 @@ class webosTvAccessory {
         if (this.infoButtonAction === undefined || this.infoButtonAction.length === 0) {
             this.infoButtonAction = 'INFO';
         }
+		this.virtualChannelList = config['virtualChannelList'];
+		if (this.virtualChannelList === undefined) {
+			this.virtualChannelList = false;
+		}
```
Comment (dkasipovic) [inline index.js]: This thing is definitely not needed, not sure why I keep copying it back

**Comment by dkasipovic** [comment]
Okay, whatever you think is best. I implemented these changes and they work for me. I will keep them up for some time to check if the channelId changes.

**Comment by PrzemyslawKlys** [comment]
This could be useful. While I renamed my buttons in HomeKit, it will require additional actions when you readd homebridge to HomeKit.

**Comment by merdok** [comment]
@PrzemyslawKlys what do you mean exactly?

**Comment by PrzemyslawKlys** [comment]
I have channels called HBO, HBO2, HBO3 and so on. But those are names in Homekit, not in Homebridge. In homebridge i have them as numbers. If i were to reset homebridge I would need to fix naming of channels again. Having names next to numbers in config (as I guess this PR is about that) would mean I wouldn't have to worry that resetting Homebridge requires any manual work from my side in Homekit.

**Comment by merdok** [comment]
The PR is about something different, but yes i was also thinking of adding a `name `property to the inputs so you can specify that. Will probably do that soon.

**Comment by merdok** [comment]
@PrzemyslawKlys optional `channelName `added in version 1.8.2
@dkasipovic optional `channelId ` property added in version 1.8.2

**Pull request closed without merging**